### ISSUE TITLE
[BUG] SDK Not Parsing Credentials.cbc File

### ISSUE THREAD

**Issue by fish-not-phish** [carbonblack/carbon-black-cloud-sdk-python#440]
**I am seeing this behaviour on: (please complete the following information):**
 - OS: Windows 11
 - Carbon Black Cloud Products: CB Enterprise EDR
 - Python Version: 3.11.6

**Describe the bug**
After creating the credentials.cbc file on Windows and giving proper permissions to the file, it cannot be read. I have installed all necessary pip packages to run this (cbapi and carbon-black-cloud-sdk).

**Steps to Reproduce**
I am actually running this from my own project I made using the CB SDK found, but here is a code snippet of the issue I am running into.

```
def global_tenant_list(path):
    try:
        with open(path, 'r') as file:
            return json.load(file)
    except FileExistsError:
        print("File not found.")
    except json.JSONDecodeError:
        print("Error decoding file.")

# defines a list of different CB profiles, in this case, tenants
global_list = None
create_global_list = input("Would you like to use a global tenant list or enter a list for every operation performed?\n1. Global List\n2. Enter a list for every operation\n Enter a number: ")
if create_global_list == "1":
    path = input("Please enter your tenant list file path: ")
    global_list = global_tenant_list(path)
    print(global_list)
elif create_global_list == "2":
    print("Proceeding without creating a global list...")
else:
    print(f"{Color.RED}Invalid number entered. Quitting...{Color.RESET}")
    sys.exit()

# enter tenant CB profile code, so it can pull the correct profile credentials from the credentials.cbc file
tenant_code = input("Enter the tenant code: ")
tenant_code = tenant_code.upper()
cb = CBCloudAPI(profile=tenant_code)
# do something
```

**Expected behavior**
I expect the credentials.cbc file to be readable.

**Screenshots**
![creds-error](https://github.com/carbonblack/carbon-black-cloud-sdk-python/assets/69283986/756af775-244e-430b-b322-887fda5fd393)

**Comment by abowersox-cb** [comment]
@fish-not-phish Can you please post those error messages _as text,_ not as a screenshot? That screenshot is _very_ hard to read.

**Comment by abowersox-cb** [comment]
Issue being tracked in Carbon Black Jira as CBAPI-5227.

**Comment by fish-not-phish** [comment]
@abowersox-cb I no longer work for the organization in which I was managing their Python API functionality. Unfortunately, because of this, I cannot reproduce the error to show the output again. I'm sure they are still interested in seeing a solution. I do offer that the screenshot is clickable and when opened in browser (if you click on it), it will expand to an easy-to-read size. I apologize for the inconvenience.

**Comment by abowersox-cb** [comment]
@fish-not-phish I _did_ click on it.  It's still really small when the screenshot pops up in a new browser window. My eyes aren't 18 years old anymore, if you know what I mean.

**Comment by fish-not-phish** [comment]
Hope this helps.

```Security warning: Windows file access to CBC SDK credentials is inherently insecure
A future version of CBC SDK will default to using a registry-based credentials provider.
Traceback (most recent call last):
	File "C:\Python311\Lib\site-packages\cbc_sdk\credential_providers\file_credential_provider.py", line 138, in get_credentials 
		parser.read(raw_cred_files)
	File "C:\Python311\Lib\configparser.py", line 713, in read
		self._read(fp, filename)
	File "C:\Python311\Lib\configparser.py", line 1101, in _read
		raise Missing Section HeaderError(fpname, lineno, line)
configparser. Missing Section HeaderError: File contains no section headers.
file: 'C:\\Users\\<redacted>\\.carbonblack\\credentials.cbc', line: 1
'ÿp[\x00c\xөөB\x00P\xooR\x00T\x00N\x001\x00]\x00\n'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
	File "\\<redacted>\Code\Scripts\Python\cb-inspector\cb_inspector.py", line 888, in <module>
		main()
	File "\\<redacted>\Code\Scripts\Python\cb-inspector\cb_inspector.py", line 386, in main
		cb = CBCloudAPI(profile=tenant_code)
		      ΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛ
	File "C:\Python311\Lib\site-packages\cbc_sdk\rest_api.py", line 49, in __init__
		super (CBCloudAPI, self).__init__(*args, **kwargs)
	File "C:\Python311\Lib\site-packages\cbc_sdk\connection.py", line 417, in _init__
		self.credentials = self.credential_provider.get_credentials(self.credential_profile_name)
					ΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛΛ
	File "C:\Python311\Lib\site-packages\cbc_sdk\credential_providers\file_credential_provider.py", line 142, in get_credentials
		raise CredentialError(f"Unable to read credential file(s) {raw_cred_files}") from e
cbc_sdk.errors.CredentialError: Unable to read credential file(s) ['C:\\Users\\<redacted>\\.carbonblack\\credentials.cbc

**Comment by abowersox-cb** [comment]
Was your `credentials.cbc` file written in UTF-16? The key part of this error message seems to be these lines:
```
configparser. Missing Section HeaderError: File contains no section headers.
file: 'C:\\Users\\<redacted>\\.carbonblack\\credentials.cbc', line: 1
'ÿp[\x00c\xөөB\x00P\xooR\x00T\x00N\x001\x00]\x00\n'
```
Those two characters at the front look like a Unicode byte-order mark (U+FEFF, in little-endian order), and the way the other characters are interleaved makes me think that I'm looking at 16-bit characters.  I'm pretty sure the `configparser` code in Python's standard library (which is what `FileCredentialsProvider` relies on) only handles UTF-8.

**Comment by fish-not-phish** [comment]
@abowersox-cb let me connect with my old team and see if they can get me the answer. I wouldn't know off the top of my head, it's been a few months.

**Comment by abowersox-cb** [comment]
This was handled in internal ticket CBAPI-5227, and will be released with SDK 1.5.2.
